packages/langium/src/validation/document-validator.ts

```diff
@@ -245,8 +248,26 @@ export function toDiagnosticSeverity(severity: 'error' | 'warning' | 'info' | 'h
     }
 }
 
+export function toDiagnosticData(severity: 'error' | 'warning' | 'info' | 'hint'): DiagnosticData {
+    switch (severity) {
+        case 'error':
+            return diagnosticData(DocumentValidator.LexingError);
+        case 'warning':
+            return diagnosticData(DocumentValidator.LexingWarning);
+        case 'info':
+            return diagnosticData(DocumentValidator.LexingInfo);
+        case 'hint':
+            return diagnosticData(DocumentValidator.LexingHint);
```

Comment:
I personally prefer a `Record<DiagnosticSeverity, DocumentValidator>` over a `switch`, and also using an enum instead of a namespace (to make this type possible)

Reply:
I aligned this with the `toDiagnosticSeverity` function in the same file so I'd say we should at least be consistent even though I do not have any hard feelings regarding either way.